### ISSUE TITLE
Automate git merge conflict resolution

### ISSUE THREAD

**Issue by tiagoefreitas** [paul-gauthier/aider#800]
### Issue

This would be awesome to merge branches or PRs.
I would just approve the pre-made merge in the vs code merge editor, if that's possible (save merge decisions in VS Code compatible format without actually commiting the merge).

The LLM would resolve the conflicts based on the functionality of the commits messages where history diverged and code understanding.

For example, I have a branch where a bunch of apis handlers were refactored to use TRPC server actions and some auth middleware upgrades. 
But that branch has diverged a lot in terms of functionality. 
There are conflicts, but understanding that branch and all its commits purpose, you know you want to keep the functionality of the main branch and just use the merge to refactor the way the apis are implemented, not its content, except the auth.

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

If you have a file full of merge conflicts, you might try adding it to aider and asking it to fix them?

**Comment by tiagoefreitas** [comment]
good idea, if that works then just need a way to auto add all conflict
files to aider context

a dedicated tool could have better UX because i would like to review and
approve each resolution step by step

workaround maybe a command line that opens the git conflicts in vs code and
the extension ads them aider

could also be helpful to add all files with lint errors/warnings for that
use case

**Comment by Kreijstal** [comment]
```
Unable to commit: Cmd('git') failed due to: exit code(128)
  cmdline: git commit -m style: run linter --no-verify -- /home/kreijstal/git/OpenVAF/openvaf/osdi/src/load.rs
  stderr: 'fatal: cannot do a partial commit during a merge.'
```
seems aider can't do merge conflicts yet

**Comment by rs545837** [comment]
For what it's worth, I built a Git merge driver called [weave](https://github.com/ataraxy-labs/weave) that tackles part of this problem. It uses tree-sitter to parse code and merge at the function/class level instead of lines. This handles the common case where git produces false conflicts because two branches added to the same line range (different functions, same location).

It doesn't do LLM-based semantic resolution (yet), but it eliminates most false positives. 31/31 clean merges vs git's 15/31 on concurrent edits. Plugs in as a standard git merge driver, works transparently with `git merge`/`rebase`/`pull`.
